storage/innobase/srv/srv0start.cc

```diff
@@ -2950,15 +2950,21 @@ struct metadata_applier {
   @param[in]      table   table to visit */
   void operator()(dict_table_t *table) const {
     ut_ad(dict_sys->dynamic_metadata != nullptr);
+#ifndef XTRABACKUP
     uint64_t autoinc = table->autoinc;
+#endif /* XTRABACKUP */
     dict_table_load_dynamic_metadata(table);
```

Comment:
we can make it simpler by incrementing right after dict_table_load_dynamic_metadata()
#ifdef XTRABACKUP
/*We don't open table by ha_innobase::open. So advance the counter for PXB
opened tables */
++table->autoinc
return;
#endif /*XTRABACKUP */

Reply:
will this be done only once? :thinking:

Reply:
I tried 2 scenarios normal insert and one where rollback is required(begin; insert into t1 values(null); )
 In both of the cases, it was done once. not sure if we have a case where will be called again. 
 but I added the case to check upper limit